### PR TITLE
refactor: improve initialization of combinatorial polyhedron

### PR DESCRIPTION
**Pull request by kliem**
### :books: Description

Motivated by #34773.

The initialization of `CombinatorialPolyhedron` is rather complicated. Changing functionality is difficult this way, so we extract some of the initialization to dedicated methods.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/geometry/polyhedron/combinatorial_polyhedron/base.pyx [hunk 1]
```diff
@@ -360,66 +360,92 @@ cdef class CombinatorialPolyhedron(SageObject):
...
-            data = data.incidence_matrix()
-
-            # Delete equations
-            if P.n_equations():
-                data_modified = data.delete_columns([e.index() for e in P.equations()])
-            else:
-                data_modified = data
+            self._init_from_polyhedron(data)
+            return
         elif isinstance(data, LatticePolytopeClass):
```
Comment (mkoeppe) [inline src/sage/geometry/polyhedron/combinatorial_polyhedron/base.pyx]: if each branch ends with `return`, perhaps better to use `if` instead of `elif`?

Comment (kliem) [reply src/sage/geometry/polyhedron/combinatorial_polyhedron/base.pyx]: Done in 5c21a7947c543159d66a3ac7a0c0c871e3bd3241.

**Review by mkoeppe — APPROVED** [review]
Nice refactoring, LGTM.

**Comment by kliem** [comment]
Thanks for the review.

**Pull request merged**